samples/openapi3/client/petstore/python-experimental/tests/test_http_signature.py

```diff
@@ -0,0 +1,370 @@
+# coding: utf-8
+
+# flake8: noqa
+
+"""
+Run the tests.
+$ docker pull swaggerapi/petstore
+$ docker run -d -e SWAGGER_HOST=http://petstore.swagger.io -e SWAGGER_BASE_PATH=/v2 -p 80:8080 swaggerapi/petstore
+$ pip install nose (optional)
+$ cd petstore_api-python
+$ nosetests -v
+"""
+
+from collections import namedtuple
+from datetime import datetime, timedelta
+import json
+import os
+import re
+import shutil
+import unittest
+from Crypto.PublicKey import RSA
+from Crypto.PublicKey import ECC
+
+import petstore_api
+from petstore_api import Configuration, signing
+from petstore_api.rest import (
+    RESTClientObject,
+    RESTResponse
+)
+
+import six
+
+from petstore_api.exceptions import (
+    ApiException,
+    ApiValueError,
+    ApiTypeError,
+)
+
+from .util import id_gen
+
+import urllib3
+
+if six.PY3:
+    from unittest.mock import patch
+else:
+    from mock import patch
+
+HOST = 'http://localhost/v2'
+
+# Test RSA private key as published in Appendix C 'Test Values' of
+# https://www.ietf.org/id/draft-cavage-http-signatures-12.txt
+RSA_TEST_PRIVATE_KEY = """-----BEGIN RSA PRIVATE KEY-----
+MIICXgIBAAKBgQDCFENGw33yGihy92pDjZQhl0C36rPJj+CvfSC8+q28hxA161QF
+NUd13wuCTUcq0Qd2qsBe/2hFyc2DCJJg0h1L78+6Z4UMR7EOcpfdUE9Hf3m/hs+F
+UR45uBJeDK1HSFHD8bHKD6kv8FPGfJTotc+2xjJwoYi+1hqp1fIekaxsyQIDAQAB
+AoGBAJR8ZkCUvx5kzv+utdl7T5MnordT1TvoXXJGXK7ZZ+UuvMNUCdN2QPc4sBiA
+QWvLw1cSKt5DsKZ8UETpYPy8pPYnnDEz2dDYiaew9+xEpubyeW2oH4Zx71wqBtOK
+kqwrXa/pzdpiucRRjk6vE6YY7EBBs/g7uanVpGibOVAEsqH1AkEA7DkjVH28WDUg
+f1nqvfn2Kj6CT7nIcE3jGJsZZ7zlZmBmHFDONMLUrXR/Zm3pR5m0tCmBqa5RK95u
+412jt1dPIwJBANJT3v8pnkth48bQo/fKel6uEYyboRtA5/uHuHkZ6FQF7OUkGogc
+mSJluOdc5t6hI1VsLn0QZEjQZMEOWr+wKSMCQQCC4kXJEsHAve77oP6HtG/IiEn7
+kpyUXRNvFsDE0czpJJBvL/aRFUJxuRK91jhjC68sA7NsKMGg5OXb5I5Jj36xAkEA
+gIT7aFOYBFwGgQAQkWNKLvySgKbAZRTeLBacpHMuQdl1DfdntvAyqpAZ0lY0RKmW
+G6aFKaqQfOXKCyWoUiVknQJAXrlgySFci/2ueKlIE1QqIiLSZ8V8OlpFLRnb1pzI
+7U1yQXnTAEFYM560yJlzUpOb1V4cScGd365tiSMvxLOvTA==
+-----END RSA PRIVATE KEY-----"""
```

Comment:
Silly question but:
Can you verify that this RSA key is was generated only for this testing?

Optionally, how about:
- generate the rsa key on the fly with:
```
from os import chmod
from Crypto.PublicKey import RSA

key = RSA.generate(2048)
with open("/tmp/private.key", 'wb') as content_file:
    chmod("/tmp/private.key", 0600)
    content_file.write(key.exportKey('PEM'))
pubkey = key.publickey()
with open("/tmp/public.key", 'wb') as content_file:
    content_file.write(pubkey.exportKey('OpenSSH'))
```
per https://stackoverflow.com/questions/2466401/how-to-generate-ssh-key-pairs-with-python
- generating the rsa key on the fly by invoking `ssh-keygen`
- or moving this data to a file in a testfiles folder like we do for other file assets https://github.com/OpenAPITools/openapi-generator/tree/master/samples/client/petstore/python-experimental/testfiles

Reply:
> Silly question but:
> Can you verify that this key is was generated only for this testing?

I have put a code comment that says it's the test key from the http signature specification. The spec has a section about "test values", with the input key and expected Authorization header. The idea was to make it possible to compare that the implementation matches exactly the spec. Though subsequently I realized it's currently not possible because there is a warning that the test values are going to be regenerated in the next draft, and the test values don't actually show the signature string.

> 
> Optionally, how about:
> 
> * generate the rsa key on the fly with:
> 
> ```
> from os import chmod
> from Crypto.PublicKey import RSA
> 
> key = RSA.generate(2048)
> with open("/tmp/private.key", 'wb') as content_file:
>     chmod("/tmp/private.key", 0600)
>     content_file.write(key.exportKey('PEM'))
> pubkey = key.publickey()
> with open("/tmp/public.key", 'wb') as content_file:
>     content_file.write(pubkey.exportKey('OpenSSH'))
> ```
> 
> per https://stackoverflow.com/questions/2466401/how-to-generate-ssh-key-pairs-with-python
> 
> * generating the rsa key on the fly by invoking `ssh-keygen`
> * or moving this data to a file in a testfiles folder like we do for other file assets https://github.com/OpenAPITools/openapi-generator/tree/master/samples/client/petstore/python-experimental/testfiles

Reply:
I think I can mark this as resolved. Feel free to reopen.